nearswap/src/twap.rs

```diff
@@ -0,0 +1,189 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::json_types::{U128, U64};
+use near_sdk::collections::{LookupMap, Vector};
+use near_sdk::serde::{Deserialize, Serialize};
+use near_sdk::{AccountId, Balance, Timestamp};
+use std::convert::{TryFrom,TryInto};
+
+use std::fmt;
+use crate::constants::*;
+
+#[derive(Clone, BorshSerialize, BorshDeserialize, Serialize, Deserialize, Copy)]
+pub struct Twap {
+    // timestamp
+    pub block_timestamp: U64,
+    // Number of observations till block_timestamp
+    pub num_of_observations: U128,
+    // cumulative price of token0 till block_timestamp
+    pub price0cumulative: U128,
```

Comment:
```suggestion
    pub price1_cumulative: U128,
```

Reply:
don't commit this suggestion - it's easier if you do rename with your editor.

Reply:
ok